pkg/controllers/node/termination/controller.go

```diff
@@ -157,6 +164,34 @@ func (c *Controller) removeFinalizer(ctx context.Context, n *v1.Node) error {
 	return nil
 }
 
+func (c *Controller) taintExpiredNode(ctx context.Context, node *v1.Node) (*time.Time, error) {
```

Comment:
nit: This is still a nit so you can take it or leave it but the intention of my original comment was to move the nodeExpirationTime function call up into finalize and pass the timestamp down to the taint function (or could taint inline in finalize). It's still unintuitive to me for the `taintExpiredNode` function to return the expiration timestamp. Again, this is a nit not a blocking issue.

Reply:
Ah I misunderstood. I should be able to do that easily.